File: 3163 String Compression III/3163.py
class Solution:
    def compressedString(self, word: str) -> str:
        ans = ""
        
        n = len(word)
        i = 0
        
        while i < n:
            cnt = 1
            c = word[i]

            while cnt < 9 and i+1 < n and word[i+1] == c:
                i += 1
                cnt += 1

            ans += str(cnt)
            ans += c
            i += 1

        return ans

File: 3163 String Compression III/3163.cpp
class Solution {
public:
    string compressedString(string word) {
        string ans = "";
        
        int n = word.size();

        for (int i = 0; i<n; i++){
            int cnt = 1;
            char c = word[i];

            while (cnt < 9 && i+1 < n && word[i+1] == c){
                i++;
                cnt++;
            } 

            ans += to_string(cnt);
            ans += c;
        }

        return ans;
    }
};
